backend/spec/features/admin/promotion_adjustments_spec.rb

```diff
@@ -240,6 +240,28 @@
       expect(first_action.calculator.preferred_amount).to eq(5)
     end
 
+    it "should not allow an Discount Rule to be added without selecting an option" do
+      fill_in "Name", with: "SAVE SAVE SAVE"
+      choose "Apply to all orders"
+      click_button "Create"
+      expect(page).to have_title("SAVE SAVE SAVE - Promotions")
+
+      within('#rule_fields') { click_button "Add" }
+      message = page.find("#promotion_rule_type").native.attribute("validationMessage")
+      expect(message).to eq "Please select an item in the list."
+    end
+
+    it "should not allow an Adjusment type to be added without selecting an option" do
+      fill_in "Name", with: "SAVE SAVE SAVE"
+      choose "Apply to all orders"
+      click_button "Create"
+      expect(page).to have_title("SAVE SAVE SAVE - Promotions")
+
```

Comment:
The 4 lines above here are the same in the previous test. What about grouping these tests inside a `context` or `describe` block and putting these lines into a `before` block? It slightly DRYes up the file and makes it evident that this part is a prerequisite for the actual tests.